python/python/bystro/proteomics/annotation_interface.py

```diff
@@ -0,0 +1,188 @@
+"""Query an annotation file and return a list of sample_ids and genes meeting the query criteria."""
+import logging
+import math
+from dataclasses import dataclass
+from typing import Any
+
+import numpy as np
+import pandas as pd
+from bystro.utils.config import get_opensearch_config
+from opensearchpy import OpenSearch
+
+logger = logging.getLogger(__file__)
+
+OPENSEARCH_CONFIG = get_opensearch_config()
+HETEROZYGOTE_DOSAGE = 1
+HOMOZYGOTE_DOSAGE = 2
+MISSING_GENO_DOSAGE = np.nan
+ONE_DAY = "1d"  # default keep_alive time for opensearch point in time index
+
+# The fields to return for each variant matched by the query
+OUTPUT_FIELDS = ["refSeq.name2", "homozygotes", "heterozygotes", "missingGenos"]
+
+
+@dataclass
+class OpenSearchQueryConfig:
+    """Represent parameters for configuring OpenSearch queries."""
+
+    max_query_size: int = 10_000
+    max_slices: int = 1024
+    keep_alive: str = ONE_DAY
+
+
+OPENSEARCH_QUERY_CONFIG = OpenSearchQueryConfig()
+
+
+def _flatten(xs: Any) -> list[Any]:  # noqa: ANN401 (`Any` is really correct here)
+    """Flatten an arbitrarily nested list."""
+    if not isinstance(xs, list):
+        return [xs]
+    return sum([_flatten(x) for x in xs], [])
+
+
+def _get_samples_genes_dosages_from_hit(hit: dict[str, Any]) -> pd.DataFrame:
+    """Given a document hit, return a dataframe of samples, genes and dosages."""
+    source = hit["_source"]
+    gene_names = _flatten(source["refSeq"]["name2"])
+    # homozygotes, heterozygotes may not be present in response, so
+    # represent them as empty lists if not.
+    heterozygotes = _flatten(source.get("heterozygotes", []))
+    homozygotes = _flatten(source.get("homozygotes", []))
```

Comment:
I would drop flatten here; this will never be a nested list, either undefined or 1D array